jsonutil/safejson/safejson.go

```diff
@@ -0,0 +1,194 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package safejson
+
+import (
+	"fmt"
+	"strconv"
+	"unicode"
+	"unicode/utf16"
+	"unicode/utf8"
+
+	"github.com/snapcore/snapd/strutil"
+)
+
+// String accepts any valid JSON string
+type String struct {
+	s string
+}
+
+func (str *String) UnmarshalJSON(in []byte) (err error) {
+	str.s, err = unmarshal(in, uOpt{})
+	return
+}
+
+// Clean returns the string, with any control characters stripped.
+func (str String) Clean() string {
```

Comment:
I suggested to make this String() below, let me know if this is insane in good or bad way please.